src/oapv_metadata.c

```diff
@@ -180,6 +180,97 @@ static void meta_free_md(oapv_md_t *md)
     }
 }
 
+int oapvm_write_mdcv(oapvm_payload_mdcv_t *mdcv, void *data, int *size)
+{
+    int i, t;
+    u32 tu32;
+    oapv_bs_t bs;
+    oapv_bsw_init(&bs, data, 24, NULL); // MDCV payload has 24 bytes
+
```

Comment:
for(i = 0; i < 3; i++) {
        t = mdcv->primary_chromaticity_x[i];
        oapv_assert_rv(t >=0 && t <= 0xFFFF, OAPV_ERR_INVALID_ARGUMENT);
        oapv_bsw_write(&bs, t, 16);

        t = mdcv->primary_chromaticity_y[i];
        oapv_assert_rv(t >=0 && t <= 0xFFFF, OAPV_ERR_INVALID_ARGUMENT);
        oapv_bsw_write(&bs, t, 16);
    }